realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -158,10 +144,25 @@ void RealSenseNodeFactory::onInit()
 		}
 		else
 		{
-			std::function<void(rs2::event_information&)> change_device_callback_function = [this](rs2::event_information& info){change_device_callback(info);};
-			_ctx.set_devices_changed_callback(change_device_callback_function);
 			privateNh.param("initial_reset", _initial_reset, false);
 			_device = getDevice();
+			if (_device && _initial_reset)
+			{
+				_initial_reset = false;
+				try
+				{
+					ROS_INFO("Resetting device...");
+					_device.hardware_reset();
+					std::this_thread::sleep_for (std::chrono::seconds(10));
```

Comment:
Is this `seconds(10)` the time that the entire Realsense node halts for? If so I'm afraid I can imagine there are applications where this change may not be desirable.

I'm not sure though, a few recommendations:
- Make it configurable (from `launch` etc.)
- If we know minimum length of period that the driver safely completes resetting with, we can set that as a default.

Reply:
Agree with @130s.  I would make this a launch parameter (can be done in the future).

Reply:
Instead, would it be advisable that I check let's say every second after the reset to see if its finished?

The upstream implementation of this is using interrupts, but there is a problem with that, where only 1 camera is detected after a reset, inside the interrupt routine

Reply:
I guess user should have a control over the recovery time.

> would it be advisable that I check let's say every second after the reset to see if its finished?

I think that's nice if it works (does the reset function return something reliably?). In this case the user would be able to pass the maximum trial time etc.?

Reply:
Good point! I would have to look if the reset function returns something useful. If not I think we can still have the maximum wait time configurable by the user, while internally it resets and keeps checking every second, until it either finds the reconnected realsense, or timesout

Reply:
That can be addressed later. We'll track the enhancement.